### PR TITLE
feat(HXFormControLElement): centralize custom form control logic

### PR DESCRIPTION
**Pull request by CITguy**
- handles adding 'hx-changed' and 'hx-touched' attrs to communicate state

JIRA: SURF-1593, SURF-1594, SURF-1595

### LGTM's
- [x] Dev LGTM
- [x] Zoom LGTM

### CODE REVIEW

**Comment by HelixUI** [comment]
Deploy preview for *helix-ui* processing.

Building with commit eb6512b3b63bd10b57cc229ffde3fa4d694ce899

https://app.netlify.com/sites/helix-ui/deploys/5c93b97e4f23910007453e6d

File: src/helix-ui/elements/HXCheckboxControlElement.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    static get is () {
+        return 'hx-checkbox-control';
+    }
+
+    /**
+     * Fetch the first `<input type="checkbox">` descendant
+     *
+     * @override
+     * @readonly
+     * @type {?HTMLElement}
```
Comment (CITguy) [inline src/helix-ui/elements/HXCheckboxControlElement.js]: `HTMLInputElement`

File: src/helix-ui/elements/HXRadioControlElement.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    static get is () {
+        return 'hx-radio-control';
+    }
+
+    /**
+     * Fetch the first `<input type="radio">` descendant
+     *
+     * @override
+     * @readonly
+     * @type {?HTMLElement}
```
Comment (CITguy) [inline src/helix-ui/elements/HXRadioControlElement.js]: `HTMLInputElement`

File: src/helix-ui/elements/HXRadioElement.html [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<!--
+  TODO: update with SVG to rener the radio button
```
Comment (CITguy) [inline src/helix-ui/elements/HXRadioElement.html]: `render`

File: src/helix-ui/elements/HXTextControlElement.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    /**
+     * Fetch the first text `<input>` descendant,
+     * whether implicit (`<input />`) or explicit
+     * (`<input type="text" />`).
+     *
+     * @override
+     * @readonly
+     * @type {?HTMLElement}
```
Comment (CITguy) [inline src/helix-ui/elements/HXTextControlElement.js]: `HTMLInputElement`

File: src/helix-ui/elements/HXTextareaControlElement.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    static get is () {
+        return 'hx-textarea-control';
+    }
+
+    /**
+     * Fetch the first text `<textarea>` descendant
+     *
+     * @override
+     * @readonly
+     * @type {?HTMLElement}
```
Comment (CITguy) [inline src/helix-ui/elements/HXTextareaControlElement.js]: `HTMLTextareaElement`

**Pull request merged**